packages/rum-core/src/performance-monitoring/metrics.js

```diff
@@ -174,18 +175,36 @@ export function calculateTotalBlockingTime(longtaskEntries) {
 }
 
 /**
- * Calculates Cumulative Layout Shift using
+ * Calculates Cumulative Layout Shift through the max-session-gap1s-limit5s technique using
  * 'layout-shift' entries captured through PerformanceObserver.
  * https://developer.mozilla.org/en-US/docs/Web/API/LayoutShift
- * https://web.dev/cls/
+ * https://web.dev/evolving-cls/
+ *
+ * We use clsState to keep track of previous layout shift events since the used technique requires
+ * to have them into account to perform the proper calculation
  */
-export function calculateCumulativeLayoutShift(clsEntries) {
+export function calculateCumulativeLayoutShift(clsState, clsEntries) {
```

Comment:
Would actually put this state inside the metrics.cls to keep it simpler, we do something similar to TBT calc. Would reduce no of fields getting added to global state.

Reply:
I agree, it is better to take advantage of the state that we already have in `metrics.js` file 👍